QUIC macOS support (#43605)
dotnet/docs: docs/fundamentals/networking/quic/quic-overview.md

- OpenSSL 3+ or 1.1 - depends on the default OpenSSL version for the distribution version, for example, OpenSSL 3 for [Ubuntu 22](https://packages.ubuntu.com/jammy/openssl) and OpenSSL 1.1 for [Ubuntu 20](https://packages.ubuntu.com/focal/utils/openssl).

- libnuma1

### macOS

QUIC is now partially supported on macOS through a non-standard Homebrew package manager with some limitations. You can install `libmsquic` on macOS using Homebrew with the following command:

```bash
brew install libmsquic
```

> To run a .NET application and ensure it uses `libmsquic`, you need to set the environment variable before running the application. This ensures the application can find the `libmsquic` library during runtime dynamic loading. You can do this by adding the following command before your main command:

Reviewer: Wording NIT:
```suggestion
To run a .NET application that uses `libmsquic`, you need to set the environment variable before running it. This ensures the application can find the `libmsquic` library during runtime dynamic loading. You can do this by adding the following command before your main command:
```
